fontforgeexe/multidialog.c

```diff
@@ -0,0 +1,642 @@
+            }
+            mpp->listcid++;
+            flarray[fl++] = &gcd[g++];
+        }
+    } else if (qstn->type == mdq_openpath || qstn->type == mdq_savepath) {
+        int fb = 0;
+        GGadgetCreateData **fbox = calloc(3, sizeof(GGadgetCreateData *));
+        mpp->memlist = g_list_append(mpp->memlist, fbox);
+        label[l].text = (unichar_t *) qstn->dflt;
+        label[l].text_is_1byte = true;
+        gcd[g].gd.label = &label[l++];
+        gcd[g].gd.mnemonic = qmne;
+        gcd[g].gd.flags = gg_enabled | gg_visible;
+        gcd[g].gd.cid = mpp->fbpair + CID_BF_PAIR_START;
+        gcd[g].data = qstn;
+        gcd[g].creator = GTextFieldCreate;
+        fbox[fb++] = &gcd[g++];
+
+        label[l].text = (unichar_t *) _("...");
+        label[l++].text_is_1byte = true;
+        gcd[g].gd.label = &label[l - 1];
+        gcd[g].gd.flags = gg_enabled | gg_visible;
+        gcd[g].gd.cid = mpp->fbpair + 1 + CID_BF_PAIR_START;
+        gcd[g].gd.handle_controlevent = Multi_BrowseFile;
+        gcd[g].data = qstn->filter;
+        mpp->fbpair += 2;
+        gcd[g].creator = GButtonCreate;
+        fbox[fb++] = &gcd[g++];
+
+        gcd[g].gd.flags = gg_enabled | gg_visible;
+        gcd[g].gd.u.boxelements = fbox;
+        gcd[g].creator = GHBoxCreate;
+        struct multi_expand *me = calloc(1, sizeof(struct multi_expand));
+        mpp->expands = g_list_append(mpp->expands, me);
+        me->gcd = &gcd[g];
+        me->is_row = false;
+        me->loc = 0;
+        flarray[fl++] = &gcd[g++];
+    }
+    gcd[g].gd.flags = gg_enabled | gg_visible | gg_flow_ocenter | gg_flow_lvcenter;
+    if (!(qstn->type == mdq_choice && qstn->checks)) {
+        gcd[g].gd.flags |= gg_flow_expand;
+    }
+    if (!qstn->align) {
+        gcd[g].gd.flags |= gg_flow_noalignlabel;
+    }
+    gcd[g].gd.u.boxelements = flarray;
+    gcd[g].creator = GFlowBoxCreate;
+    gcd[g].gd.label = (GTextInfo *) qgcd;
+
+    return &gcd[g];
+}
+
+static GGadgetCreateData *LayoutMultiDlgCategoryBody(MultiDlgCategory *cat, struct multi_postproc *mpp) {
+    int gcnt = 1, flcnt = 1, g = 0, l = 0, fl = 0, i;
+    GGadgetCreateData *gcd, **s1barray;
+
+    s1barray = calloc(cat->len + 1, sizeof(GGadgetCreateData *));
+    mpp->memlist = g_list_append(mpp->memlist, s1barray);
+    for (i = 0; i < cat->len; ++i) {
+        s1barray[i] = LayoutMultiDlgQuestion(cat->questions + i, mpp);
+        if (s1barray[i] == NULL) {
+            return false;
+        }
+    }
+
+    // Allocate 2 elements in case this is created at the top of a tabset,
+    // which calls GGadgetsCreate()
+    gcd = calloc(2, sizeof(GGadgetCreateData));
+    mpp->memlist = g_list_append(mpp->memlist, gcd);
+    gcd[0].gd.flags = gg_enabled | gg_visible | gg_s1_vert | gg_s1_flowalign | gg_s1_expand;
+    gcd[0].gd.u.boxelements = s1barray;
+    gcd[0].creator = GScroll1BoxCreate;
+
+    return gcd;
+}
+
+static void MultiSyncChoices(MultiDlgSpec *spec) {
```

Comment:
Is this name descriptive? It seems to me that the function just sets defaults.

Reply:
I'll change it to `MultiSetDefaults()` then.

Reply:
Changed the function name